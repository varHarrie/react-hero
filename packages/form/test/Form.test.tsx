import 'jest'

import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { configure, render } from 'enzyme'

import { Form, FormField, FormItem, FormStore } from '..'

configure({ adapter: new Adapter() })

describe('Form', () => {
  it('should render correctly', () => {
    const wrapper = render(<Form store={new FormStore()} />)
    expect(wrapper).toMatchSnapshot()

    const wrapper2 = render(<Form store={new FormStore()}>Hello World</Form>)
    expect(wrapper2).toMatchSnapshot()

    const wrapper3 = render(
      <Form store={new FormStore()}>
        <div>Hello</div>
        <div>World</div>
      </Form>
    )
    expect(wrapper3).toMatchSnapshot()
  })
})

describe('Field', () => {
  it('should render correctly', () => {
    const wrapper = render(
      <FormField label='Label' labelWidth={200} gutter={30} suffix='Suffix' required inline>
        <input type='text' />
      </FormField>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render error correctly', () => {
    const store = new FormStore()
    store.error('Name', 'Error Message')

    const wrapper = render(
      <Form store={store}>
        <div>
          <FormField label='Label' name='Name' />
        </div>
      </Form>
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Item', () => {
  it('should render correctly', () => {
    const store = new FormStore()

    const wrapper = render(
      <Form store={store}>
        <FormItem>
          <input type='text' />
        </FormItem>
      </Form>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
