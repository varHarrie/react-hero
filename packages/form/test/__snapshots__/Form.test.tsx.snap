// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should render correctly 1`] = `
<div
  class="rh-form-field rh-form-field--inline rh-form-field--required "
>
  <div
    class="rh-form-field__header"
    style="width:200px;margin-right:30px"
  >
    Label
  </div>
  <div
    class="rh-form-field__container"
  >
    <div
      class="rh-form-field__control"
    >
      <input
        type="text"
      />
    </div>
    <div
      class="rh-form-field__message"
    />
  </div>
  <div
    class="rh-form-field__footer"
  >
    Suffix
  </div>
</div>
`;

exports[`Field should render error correctly 1`] = `
<form
  class="rh-form "
>
  <div>
    <div
      class="rh-form-field rh-form-field--error "
    >
      <div
        class="rh-form-field__header"
      >
        Label
      </div>
      <div
        class="rh-form-field__container"
      >
        <div
          class="rh-form-field__control"
        />
        <div
          class="rh-form-field__message"
        >
          Error Message
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Form should render correctly 1`] = `
<form
  class="rh-form "
/>
`;

exports[`Form should render correctly 2`] = `
<form
  class="rh-form "
>
  Hello World
</form>
`;

exports[`Form should render correctly 3`] = `
<form
  class="rh-form "
>
  <div>
    Hello
  </div>
  <div>
    World
  </div>
</form>
`;

exports[`Item should render correctly 1`] = `
<form
  class="rh-form "
>
  <input
    type="text"
  />
</form>
`;
